src/sources/opentelemetry/tests.rs

```diff
@@ -334,6 +341,337 @@ async fn receive_grpc_logs_legacy_namespace() {
+                keepalive: Default::default(),
+                headers: vec![],
+                host_key: OptionalValuePath::new("remote_ip"),
+            },
+            acknowledgements: Default::default(),
+            log_namespace: Some(true),
+        };
+        let schema_definitions = source
+            .outputs(LogNamespace::Vector)
+            .remove(0)
+            .schema_definition(true);
+
+        let (sender, logs_output, _) = new_source(EventStatus::Delivered, LOGS.to_string());
+        let server = source
+            .build(SourceContext::new_test(sender, None))
+            .await
+            .unwrap();
+        tokio::spawn(server);
+        test_util::wait_for_tcp(http_addr).await;
+
+        let client = reqwest::Client::new();
+        let req = ExportLogsServiceRequest {
+            resource_logs: vec![ResourceLogs {
+                resource: None,
+                scope_logs: vec![ScopeLogs {
+                    scope: None,
+                    log_records: vec![LogRecord {
+                        time_unix_nano: 1,
+                        observed_time_unix_nano: 2,
+                        severity_number: 9,
+                        severity_text: "info".into(),
+                        body: Some(AnyValue {
+                            value: Some(any_value::Value::StringValue("log body".into())),
+                        }),
+                        attributes: vec![],
+                        dropped_attributes_count: 3,
+                        flags: 4,
+                        // opentelemetry sdk will hex::decode the given trace_id and span_id
+                        trace_id: str_into_hex_bytes("4ac52aadf321c2e531db005df08792f5"),
+                        span_id: str_into_hex_bytes("0b9e4bda2a55530d"),
+                    }],
+                    schema_url: "v1".into(),
+                }],
+                schema_url: "v1".into(),
+            }],
+        };
+        let _res = client
+            .post(format!("http://{}/v1/logs", http_addr))
+            .header("Content-Type", "application/x-protobuf")
+            .body(req.encode_to_vec())
+            .send()
+            .await
+            .expect("Failed to send log to Opentelemetry Collector.");
+
+        let mut output = test_util::collect_ready(logs_output).await;
+        assert_eq!(output.len(), 1);
+        let event = output.pop().unwrap();
+        schema_definitions.unwrap().assert_valid_for_event(&event);
+
+        assert_eq!(event.as_log().get(".").unwrap(), &value!("log body"));
+
+        let meta = event.as_log().metadata().value();
+        assert_eq!(
+            meta.get(path!("vector", "source_type")).unwrap(),
+            &value!(OpentelemetryConfig::NAME)
+        );
+        assert!(meta
+            .get(path!("vector", "ingest_timestamp"))
+            .unwrap()
+            .is_timestamp());
+        assert_eq!(
+            meta.get(path!("opentelemetry", "host")).unwrap(),
```

Comment:
I would expect this to be `remote_ip`. I think this is also broken in `http_server` because we hardcode `path!("host")` when enriching events.

Reply:
I also wondered this, but I assumed it was intentional that - when using log namespacing - the source IP was always under `path!("host")` and that the `host_key` configuration was only for the legacy namespace.

But you bring up some good questions about the behavior of this setting, so I reverted the `host_key` change from this PR.